components/ConvertSteps/Converter.js

```diff
@@ -0,0 +1,195 @@
+        },
+      }
+    })
+  }, [])
+
+  const handleClaimOrderStep = useCallback(
+    async buyOrderHash => {
+      const stepType = 'claimOrder'
+
+      setCurrentStep(stepType)
+
+      try {
+        // Awaiting confirmation
+        applyStepState(stepType, 'waiting')
+        const transaction = await claimOrder(buyOrderHash, toAnj)
+
+        // Mining transaction
+        applyStepState(stepType, 'working', transaction.hash)
+        await transaction.wait()
+
+        // Success
+        applyStepState(stepType, 'success')
+        setStepperStage('success')
+      } catch (err) {
+        applyStepState(stepType, 'error')
+        setStepperStage('error')
+        console.log(err)
+      }
+    },
+    [claimOrder, applyStepState, toAnj]
+  )
+
+  const handleBuyOrderStep = useCallback(async () => {
+    const stepType = 'buyOrder'
+
+    setCurrentStep(stepType)
+
+    try {
+      // Awaiting confirmation
+      applyStepState(stepType, 'waiting')
+      const transaction = await openOrder(amountSource, toAnj)
+
+      // Mining transaction
+      applyStepState(stepType, 'working', transaction.hash)
+      await transaction.wait()
+
+      // Success
+      applyStepState(stepType, 'success')
+      handleClaimOrderStep(transaction.hash)
+    } catch (err) {
+      applyStepState(stepType, 'error')
+      setStepperStage('error')
+      console.log(err)
+    }
+  }, [amountSource, openOrder, applyStepState, toAnj, handleClaimOrderStep])
+
+  const handleApprovalStep = useCallback(async () => {
+    const stepType = 'approval'
+
+    setCurrentStep(stepType)
+
+    try {
+      // Awaiting approval
+      applyStepState(stepType, 'waiting')
+      await checkAllowance(amountSource)
+
+      // Success
+      applyStepState(stepType, 'success')
+      handleBuyOrderStep()
+    } catch (err) {
+      applyStepState(stepType, 'error')
+      setStepperStage('error')
+      console.log(err)
+    }
+  }, [amountSource, checkAllowance, applyStepState, handleBuyOrderStep])
+
+  function handleRepeatTransaction() {
+    if (currentStep === 'approval') {
+      handleApprovalStep()
+    } else if (currentStep === 'buyOrder') {
+      handleBuyOrderStep()
+    } else if (currentStep === 'claimOrder') {
+      handleClaimOrderStep(stepState.buyOrder.hash)
+    }
+  }
+
+  useEffect(() => {
+    if (amountSource) {
+      toAnj ? handleApprovalStep() : handleBuyOrderStep()
+    }
+  }, [amountSource, handleApprovalStep, handleBuyOrderStep, toAnj])
+
+  return (
+    <StepperLayout
+      fromAmount={formattedFromAmount}
+      toAmount={formattedToAmount}
+      stage={stepperStage}
+      toAnj={toAnj}
+      onRepeatTransaction={handleRepeatTransaction}
+      onReturnHome={handleReturnHome}
+    >
+      {toAnj && (
+        <>
+          <Step
+            title="Approve ANT"
+            number="1"
+            active={stepState.approval.active}
+            status={stepState.approval.status}
```

Comment:
We should also show the transaction hash for this step, since it creates a transaction